scripts/world_bank/datasets/datasets.py

```diff
@@ -322,15 +387,159 @@ def get_codes_from_zip(zip_file):
                             code,
                         INDICATOR_NAME_KEY:
                             series_row.get(INDICATOR_NAME_KEY),
+                        NUM_DATASETS_KEY:
+                            1,
+                        TOPIC_KEY:
+                            series_row.get(TOPIC_KEY),
+                        UNIT_OF_MEASURE_KEY:
+                            series_row.get(UNIT_OF_MEASURE_KEY),
                         SHORT_DEFINITION_KEY:
                             series_row.get(SHORT_DEFINITION_KEY),
-                        TOPIC_KEY:
-                            series_row.get(TOPIC_KEY)
+                        LONG_DEFINITION_KEY:
+                            series_row.get(LONG_DEFINITION_KEY),
                     }
                 return codes
         return {}
 
 
+def write_csv(csv_file_path, csv_columns, csv_rows):
+    with open(csv_file_path, 'w', newline='') as out:
+        csv_writer = csv.DictWriter(out,
+                                    fieldnames=csv_columns,
+                                    lineterminator='\n')
+        csv_writer.writeheader()
+        csv_writer.writerows(csv_rows)
+
+
+def write_all_observations(stat_vars_file):
+    start = datetime.now()
+    logging.info('Start: %s', start)
+
+    svs = load_stat_vars(stat_vars_file)
+
+    zip_files = []
+    for file_name in os.listdir(DOWNLOADS_DIR):
+        if file_name.endswith(CSV_ZIP_FILE_SUFFIX):
+            zip_files.append(f"{DOWNLOADS_DIR}/{file_name}")
+
+    with multiprocessing.Pool(POOL_SIZE) as pool:
+        pool.starmap(write_observations_from_zip, zip(zip_files, repeat(svs)))
+
+    end = datetime.now()
+    logging.info('End: %s', end)
+    logging.info('Duration: %s', str(end - start))
+
+
+def write_observations_from_zip(zip_file, svs):
+    csv_rows = get_observations_from_zip(zip_file, svs)
+    if len(csv_rows) == 0:
+        logging.info(
+            'SKIPPED writing obs file, no observations extracted from %s',
+            zip_file)
+        return
+
+    obs_file_name = f"{zip_file.split('/')[-1].split('.')[0]}_obs.csv"
+    obs_file_path = f"{OBSERVATIONS_DIR}/{obs_file_name}"
+    logging.info('Writing %s observations from %s to %s', len(csv_rows),
+                 zip_file, obs_file_path)
+    write_csv(obs_file_path, OBS_CSV_COLUMNS, csv_rows)
+
+
+def get_observations_from_zip(zip_file, svs):
+    with zipfile.ZipFile(zip_file, 'r') as zip:
+        file_names = get_data_and_series_file_names(zip)
+        if file_names is None or len(file_names) != 2:
+            logging.warning('No data file found in ZIP file: %s', zip_file)
+            return []
+        else:
+            data_file = file_names[0]
+            # Use name of file (excluding the extension) as the measurement method
+            measurement_method = f"{WORLD_BANK_STAT_VAR_PREFIX}_{zip_file.split('/')[-1].split('.')[0]}"
```

Comment:
can do `os.path.basename(...)` instead of split('/')[-1] to get file name.

Reply:
We are getting the filename as a method parameter and not from OS here so keeping it as-is.